consensus/blake3pow/consensus.go

```diff
@@ -0,0 +1,509 @@
+	// difficulty adjustment can only be checked in zone
+	if nodeCtx == common.ZONE_CTX {
+		expected := blake3pow.CalcDifficulty(chain, parent)
+		if expected.Cmp(header.Difficulty()) != 0 {
+			return fmt.Errorf("invalid difficulty: have %v, want %v", header.Difficulty(), expected)
+		}
+	}
+
+	_, order, err := blake3pow.CalcOrder(header)
+	if err != nil {
+		return err
+	}
+	if order > nodeCtx {
+		return fmt.Errorf("order of the block is greater than the context")
+	}
+
+	if !common.NodeLocation.InSameSliceAs(header.Location()) {
+		return fmt.Errorf("block location is not in the same slice as the node location")
+	}
+
+	// Verify that the parent entropy is calculated correctly on the header
+	parentEntropy := blake3pow.TotalLogS(parent)
+	if parentEntropy.Cmp(header.ParentEntropy()) != 0 {
+		return fmt.Errorf("invalid parent entropy: have %v, want %v", header.ParentEntropy(), parentEntropy)
+	}
+
+	// If not prime, verify the parentDeltaS field as well
+	if nodeCtx > common.PRIME_CTX {
+		_, parentOrder, _ := blake3pow.CalcOrder(parent)
+		// If parent was dom, deltaS is zero and otherwise should be the calc delta s on the parent
+		if parentOrder < nodeCtx {
+			if common.Big0.Cmp(header.ParentDeltaS()) != 0 {
+				return fmt.Errorf("invalid parent delta s: have %v, want %v", header.ParentDeltaS(), common.Big0)
+			}
+		} else {
+			parentDeltaS := blake3pow.DeltaLogS(parent)
+			if parentDeltaS.Cmp(header.ParentDeltaS()) != 0 {
+				return fmt.Errorf("invalid parent delta s: have %v, want %v", header.ParentDeltaS(), parentDeltaS)
+			}
+		}
+	}
+
+	if nodeCtx == common.ZONE_CTX {
+		// Verify that the gas limit is <= 2^63-1
+		cap := uint64(0x7fffffffffffffff)
+		if header.GasLimit() > cap {
+			return fmt.Errorf("invalid gasLimit: have %v, max %v", header.GasLimit(), cap)
+		}
+		// Verify that the gasUsed is <= gasLimit
+		if header.GasUsed() > header.GasLimit() {
+			return fmt.Errorf("invalid gasUsed: have %d, gasLimit %d", header.GasUsed(), header.GasLimit())
+		}
+		// Verify the block's gas usage and verify the base fee.
+		// Verify that the gas limit remains within allowed bounds
+		if err := misc.VerifyGaslimit(parent.GasLimit(), header.GasLimit()); err != nil {
+			return err
+		}
+		// Verify the header is not malformed
+		if header.BaseFee() == nil {
+			return fmt.Errorf("header is missing baseFee")
+		}
+		// Verify the baseFee is correct based on the parent header.
+		expectedBaseFee := misc.CalcBaseFee(chain.Config(), parent)
+		if header.BaseFee().Cmp(expectedBaseFee) != 0 {
+			return fmt.Errorf("invalid baseFee: have %s, want %s, parentBaseFee %s, parentGasUsed %d",
+				expectedBaseFee, header.BaseFee(), parent.BaseFee(), parent.GasUsed())
+		}
+	}
+	// Verify that the block number is parent's +1
+	if diff := new(big.Int).Sub(header.Number(), parent.Number()); diff.Cmp(big.NewInt(1)) != 0 {
+		return consensus.ErrInvalidNumber
+	}
+	// Verify the engine specific seal securing the block
+	if seal {
```

Comment:
this `if seal {...}` block needs to be removed. It is redundant with the `CalcOrder()` call and not present in `Progpow/consensus.go`

Reply:
Removed